master/buildbot/reporters/bitbucket.py

```diff
@@ -124,26 +124,15 @@ def get_owner_and_repo(repourl):
         Takes a git repository URL from Bitbucket and tries to determine the owner and repository
         name
         :param repourl: Bitbucket git repo in the form of
-                    git@bitbucket.com:OWNER/REPONAME.git
-                    https://bitbucket.com/OWNER/REPONAME.git
-                    ssh://git@bitbucket.com/OWNER/REPONAME.git
+                    git@bitbucket.org:OWNER/REPONAME.git
+                    https://bitbucket.org/OWNER/REPONAME.git
+                    ssh://git@bitbucket.org/OWNER/REPONAME.git
+                    https://api.bitbucket.org/2.0/repositories/OWNER/REPONAME
         :return: owner, repo: The owner of the repository and the repository name
         """
-        parsed = urlparse(repourl)
+        m = re.search(r"^(?:git@|ssh://git@|https://(?:api\.)?)bitbucket\.\w+"
+                      r"(?:[:/])(?:2\.0/repositories/)?(.+)/(.+(?=\.git)|.+)",
+                      repourl)
+        assert m is not None, 'OWNER/REPONAME is expected'
 
-        if parsed.scheme:
-            path = parsed.path[1:]
-        else:
-            # we assume git@host:owner/repo.git here
-            path = parsed.path.split(':', 1)[-1]
-
-        if path.endswith('.git'):
-            path = path[:-4]
-        while path.endswith('/'):
-            path = path[:-1]
```

Comment:
I know it is not a problem in your PR, but could you replace this loop with use of `lstrip('/')`? There's also a case above where `rstrip('/')` would be useful.

Reply:
Done